site/components/map/ServicesMap.tsx

```diff
@@ -172,17 +180,52 @@ const Map = ({
                 const stop: Stop[] = getSelectedStopsFromMapMarkers(markerData, id);
                 const atcoCodes = getAtcoCodesFromSelectedStops(stop);
 
-                const servicesInPolygon = await fetchServicesByStops({
+                const servicesInPolygon: ServiceByStop[] = await fetchServicesByStops({
                     atcoCodes,
                     includeRoutes: true,
                     dataSource: dataSource,
                 });
 
                 if (servicesInPolygon.length === 0) {
-                    setWarningMessage(warningMessageText(selected.length).noServiceAssociatedWithStop);
+                    setWarningMessage(warningMessageText(selectedStop.length).noServiceAssociatedWithStop);
                     return;
                 }
 
+                const servicesWithoutDuplicates: ServiceByStop[] = filterServices(servicesInPolygon);
+
+                const servicesRoutesForGivenStop = servicesWithoutDuplicates.map((service) => {
```

Comment:
This can be done without the return statement:

```ts
const servicesRoutesForGivenStop = servicesWithoutDuplicates.map((service) => ({
    inbound: service.routes.inbound,
    outbound: service.routes.outbound,
    serviceId: service.id,
}));
```